### PR TITLE
Feature/kebab-case-validation

### LINKED ISSUE #1: Validation all files in project if do they match kebab-case pattern
Validate all filenames in project by running script using node.js

### PR DESCRIPTION
**Pull request by bigsergey**
PR resolves #1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const glob = require('fast-glob');
+
+const lintFile = require('./lint-file');
+const logErrors = require('./log-errors');
+
+glob('**/*', { ignore: ['node_modules'] })
+  .then(files => files.map(lintFile))
```
Comment (erykpiast) [inline index.js]: Single `then` is enough, steps are synchronous. Use `pipe` if you want to keep the look of the code.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  .then(files => {
+    if (files.length > 0) {
+      logErrors(files);
+      process.exit(1);
+    } else {
+      console.log('Success!!')
+      process.exit(0);
+    }
+  })
+  .catch(console.error);
```
Comment (erykpiast) [inline index.js]: explicit `process.exit(1)` would be nice here as well

File: lint-file.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const path =require('path');
+
+const kebabCase = /^[a-z0-9]*(?:[a-z0-9+-])+[a-z0-9]+$/;
```
Comment (erykpiast) [inline lint-file.js]: maybe instead coding regular expressions manually, use packages like [`camelcase`](https://www.npmjs.com/package/camelcase) and [`decamelize`](https://www.npmjs.com/package/decamelize) and test filenames like this 

```js
const isValid = decamelize(basename) === basename;
```

You'll have different styles supported almost for free.

File: lint-file.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+module.exports = function lintFile(file) {
+  const extensionName = path.extname(file);
+  const basename = path.basename(file, extensionName);
+  const isValid = kebabCase.test(basename);
+  const error = !isValid
+    ? { message: `${file} has wrong name.` }
+    : undefined;
+
+  return {
+    isValid,
```
Comment (erykpiast) [inline lint-file.js]: Isn't it redundant? If `error` is defined, the file can't be valid. Returning `Error` or `null` would be enough I think.

File: package.json [hunk 5]
```diff
@@ -21,5 +21,11 @@
   "bugs": {
     "url": "https://github.com/bigsergey/filenamelint/issues"
   },
-  "homepage": "https://github.com/bigsergey/filenamelint#readme"
+  "homepage": "https://github.com/bigsergey/filenamelint#readme",
+  "scripts": {
+    "start": "node index.js"
+  },
```
Comment (erykpiast) [inline package.json]: add `engines` field!

File: index.js (hunk 1 above)
Comment (bigsergey) [reply index.js]: which library do you recommend for pipes? ramda?

Comment (erykpiast) [reply index.js]: For now I'd start with some single-purpose package. Like [this one](https://github.com/acvos/function-pipe). If you need more similar functions, go on with Ramda.

File: package.json [hunk 6]
```diff
@@ -21,5 +21,10 @@
   "bugs": {
     "url": "https://github.com/bigsergey/filenamelint/issues"
   },
-  "homepage": "https://github.com/bigsergey/filenamelint#readme"
+  "homepage": "https://github.com/bigsergey/filenamelint#readme",
+  "dependencies": {
+    "fast-glob": "^3.1.0",
```
Comment (erykpiast) [inline package.json]: remove `^`es + add `.npmrc` with `save-exact=true`!

File: index.js (hunk 1 above)
Comment (bigsergey) [reply index.js]: done

File: lint-file.js (hunk 3 above)
Comment (bigsergey) [reply lint-file.js]: done

File: lint-file.js (hunk 4 above)
Comment (bigsergey) [reply lint-file.js]: done

File: package.json (hunk 6 above)
Comment (bigsergey) [reply package.json]: done

File: package.json (hunk 5 above)
Comment (bigsergey) [reply package.json]: done

**Pull request merged**